test/Container/ApplicationFactoryTest.php

```diff
@@ -65,6 +127,47 @@ public function getRouterFromApplication(Application $app)
         return $r->getValue($app);
     }
 
+    /**
+     * @expectedException \Zend\Expressive\Exception\InvalidArgumentException
+     */
+    public function testInvalidPhpSettingsRaisesException()
+    {
+        $config = ['php_settings' => 'invalid'];
+
+        $this->factory->__invoke($this->getContainer($config)->reveal());
+    }
+
+    /**
+     * @expectedException \Zend\Expressive\Container\Exception\PhpSettingsFailureException
+     */
+    public function testWhenSettingPhpConfigurationFailsExceptionIsRaised()
+    {
+        $config = [
+            'php_settings' => [
+                'invalid_option' => 1,
+            ],
+        ];
+
+        $this->factory->__invoke($this->getContainer($config)->reveal());
+    }
+
+    public function testFactorySetsPhpSettingsFromConfig()
+    {
+        $config = [
+            'php_settings' => [
+                'display_errors' => 1,
+                'date.timezone' => 'Europe/Belgrade',
+            ],
+        ];
+
+        $this->factory->__invoke($this->getContainer($config)->reveal());
+
+        foreach ($config['php_settings'] as $name => $value) {
+            $this->assertEquals($value, ini_get($name));
+            ini_restore($name);
+        }
```

Comment:
PHPUnit assertions throw exceptions when assertion fail. So ini_restore won't be restored when test fail so next tests in the run will be executed with a wrong environment setting.

After $config

```php
   $this->iniSet('display_errors', 0);
   $this->iniSet('date.timezone', 'UTC');
```